From 0b6a5c735649d668b2ba1ccb913558969ae8890a Mon Sep 17 00:00:00 2001
From: root <c@jia.je>
Date: Fri, 2 Feb 2024 06:06:46 -0800
Subject: [PATCH] Fix gold on mips64 targets (loongson3/loongson2f only)

gold/

2016-08-09  Aurelien Jarno  <aurelien@aurel32.net>

        * configure.tgt: Add mips64el*-*-*|mips64le*-*-* and mips64*-*-*.
---
 gold/configure.tgt | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/gold/configure.tgt b/gold/configure.tgt
index a80ec6a320..ccc17f735b 100644
--- a/gold/configure.tgt
+++ b/gold/configure.tgt
@@ -153,6 +153,13 @@ aarch64*-*)
  targ_big_endian=false
  targ_extra_big_endian=true
  ;;
+mips*64*el*-*-*|mips*64*le*-*-*)
+ targ_obj=mips
+ targ_machine=EM_MIPS_RS3_LE
+ targ_size=64
+ targ_big_endian=false
+ targ_extra_big_endian=true
+ ;;
 mips*el*-*-*|mips*le*-*-*)
  targ_obj=mips
  targ_machine=EM_MIPS
@@ -161,6 +168,13 @@ mips*el*-*-*|mips*le*-*-*)
  targ_big_endian=false
  targ_extra_big_endian=true
  ;;
+mips*64*-*-*)
+ targ_obj=mips
+ targ_machine=EM_MIPS
+ targ_size=64
+ targ_big_endian=true
+ targ_extra_big_endian=false
+ ;;
 mips*-*-*)
  targ_obj=mips
  targ_machine=EM_MIPS
-- 
2.48.0.rc2.windows.1

